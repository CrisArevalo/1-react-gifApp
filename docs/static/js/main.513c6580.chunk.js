(this["webpackJsonp02-gifexpertapp"]=this["webpackJsonp02-gifexpertapp"]||[]).push([[0],{16:function(t,e,n){},17:function(t,e,n){"use strict";n.r(e);var r=n(1),a=n.n(r),c=n(7),i=n.n(c),u=n(2),s=n(9),o=n(0),j=function(t){var e=t.setTags,n=Object(r.useState)(""),a=Object(u.a)(n,2),c=a[0],i=a[1];return Object(o.jsx)("form",{onSubmit:function(t){t.preventDefault(),c.trim().length>2&&(e((function(t){return[c].concat(Object(s.a)(t))})),i(""))},children:Object(o.jsx)("input",{type:"text",value:c,onChange:function(t){i(t.target.value)}})})},f=n(6),l=n.n(f),b=n(8),d=function(){var t=Object(b.a)(l.a.mark((function t(e){var n,r,a,c,i;return l.a.wrap((function(t){for(;;)switch(t.prev=t.next){case 0:return n="https://api.giphy.com/v1/gifs/search?q=".concat(encodeURI(e),"&limit=6&api_key=wYSQ9FXFZIjyOfD57moAlX5dTOuMLzsQ"),t.next=3,fetch(n);case 3:return r=t.sent,t.next=6,r.json();case 6:return a=t.sent,c=a.data,i=c.map((function(t){return{id:t.id,title:t.title,url:t.images.downsized_medium.url}})),t.abrupt("return",i);case 10:case"end":return t.stop()}}),t)})));return function(e){return t.apply(this,arguments)}}(),p=function(t){var e=t.url,n=(t.title,t.id);return Object(o.jsx)(o.Fragment,{children:Object(o.jsx)("img",{className:"GifGrid animate__animated animate__bounce animate__bounceInRight",src:e,alt:n})})},O=function(t){var e=t.tags,n=function(t){var e=Object(r.useState)({data:[],loading:!0}),n=Object(u.a)(e,2),a=n[0],c=n[1];return Object(r.useEffect)((function(){d(t).then((function(t){c({data:t,loading:!1})}))}),[t]),a}(e),a=n.data;return Object(o.jsxs)(o.Fragment,{children:[Object(o.jsx)("h3",{children:e}),a.map((function(t){var e=t.id,n=t.title,r=t.url;return Object(o.jsx)(p,{url:r,title:n},e)}))]})};function h(){var t=Object(r.useState)([]),e=Object(u.a)(t,2),n=e[0],a=e[1];return Object(o.jsxs)(o.Fragment,{children:[Object(o.jsx)("h2",{children:"GifExpertApp"}),Object(o.jsx)(j,{setTags:a}),Object(o.jsx)("hr",{}),Object(o.jsx)("ol",{children:n.map((function(t){return Object(o.jsx)(O,{tags:t},t)}))})]})}n(16);i.a.render(Object(o.jsx)(a.a.StrictMode,{children:Object(o.jsx)(h,{})}),document.getElementById("root"))}},[[17,1,2]]]);
//# sourceMappingURL=main.513c6580.chunk.js.map