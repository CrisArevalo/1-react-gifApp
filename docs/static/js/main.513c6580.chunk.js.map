{"version":3,"sources":["components/AddCategory.js","helpers/getGifs.js","components/GifGridItem.js","components/GifGrid.js","hooks/useFetchGifs.js","gifExpertApp.js","index.js"],"names":["AddCategory","setTags","useState","value","setValue","onSubmit","e","preventDefault","trim","length","tags","type","onChange","target","getGifs","a","url","encodeURI","fetch","resp","json","data","gif","map","img","id","title","images","downsized_medium","GifGridItem","className","src","alt","GifGrid","loading","state","setState","useEffect","then","imgs","useFetchGifs","App","t","ReactDOM","render","StrictMode","document","getElementById"],"mappings":"8MAGaA,EAAc,SAAC,GAAiB,IAAfC,EAAc,EAAdA,QAC1B,EAA0BC,mBAAS,IAAnC,mBAAOC,EAAP,KAAcC,EAAd,KAeA,OACI,sBAAMC,SAVW,SAACC,GAClBA,EAAEC,iBAEEJ,EAAMK,OAAOC,OAAS,IACtBR,GAAQ,SAACS,GAAD,OAAWP,GAAX,mBAAqBO,OAC7BN,EAAS,MAKb,SACI,uBAAOO,KAAK,OAAOR,MAAOA,EAAOS,SAff,SAACN,GACvBF,EAASE,EAAEO,OAAOV,a,uBCPbW,EAAO,uCAAG,WAAOJ,GAAP,uBAAAK,EAAA,6DACbC,EADa,iDACmCC,UAClDP,GAFe,8DAIAQ,MAAMF,GAJN,cAIbG,EAJa,gBAKIA,EAAKC,OALT,uBAKXC,EALW,EAKXA,KAEFC,EAAMD,EAAKE,KAAI,SAACC,GAClB,MAAO,CACHC,GAAID,EAAIC,GACRC,MAAOF,EAAIE,MACXV,IAAKQ,EAAIG,OAAOC,iBAAiBZ,QAXtB,kBAeZM,GAfY,4CAAH,sDCEPO,EAAc,SAAC,GAAwB,IAAtBb,EAAqB,EAArBA,IAAYS,GAAS,EAAhBC,MAAgB,EAATD,IAGtC,OACI,mCACI,qBACIK,UAAU,mEACVC,IAAKf,EACLgB,IAAKP,OCNRQ,EAAU,SAAC,GAAc,IAAZvB,EAAW,EAAXA,KACtB,ECFwB,SAACA,GACzB,MAA0BR,mBAAS,CAC/BmB,KAAM,GACNa,SAAS,IAFb,mBAAOC,EAAP,KAAcC,EAAd,KAcA,OATAC,qBAAU,WACNvB,EAAQJ,GAAM4B,MAAK,SAACC,GAChBH,EAAS,CACLf,KAAMkB,EACNL,SAAS,SAGlB,CAACxB,IAEGyB,EDbUK,CAAa9B,GAAtBW,EAAR,EAAQA,KAER,OACI,qCACI,6BAAKX,IAEJW,EAAKE,KAAI,YAAyB,IAAtBE,EAAqB,EAArBA,GAAIC,EAAiB,EAAjBA,MAAOV,EAAU,EAAVA,IACpB,OAAO,cAAC,EAAD,CAAsBA,IAAKA,EAAKU,MAAOA,GAArBD,UEF1B,SAASgB,IACpB,MAAwBvC,mBAAS,IAAjC,mBAAOQ,EAAP,KAAaT,EAAb,KAEA,OACI,qCACI,8CACA,cAAC,EAAD,CAAaA,QAASA,IACtB,uBAEA,6BACKS,EAAKa,KAAI,SAACmB,GAAD,OACN,cAAC,EAAD,CAAiBhC,KAAMgC,GAATA,W,MChBlCC,IAASC,OACL,cAAC,IAAMC,WAAP,UACI,cAACJ,EAAD,MAEJK,SAASC,eAAe,W","file":"static/js/main.513c6580.chunk.js","sourcesContent":["import React, { useState } from \"react\";\r\nimport { PropTypes } from \"prop-types\";\r\n\r\nexport const AddCategory = ({ setTags }) => {\r\n    const [value, setValue] = useState(\"\");\r\n\r\n    const handleValueChange = (e) => {\r\n        setValue(e.target.value);\r\n    };\r\n\r\n    const handleSubmit = (e) => {\r\n        e.preventDefault();\r\n\r\n        if (value.trim().length > 2) {\r\n            setTags((tags) => [value, ...tags]);\r\n            setValue(\"\");\r\n        }\r\n    };\r\n\r\n    return (\r\n        <form onSubmit={handleSubmit}>\r\n            <input type=\"text\" value={value} onChange={handleValueChange} />\r\n        </form>\r\n    );\r\n};\r\n\r\nAddCategory.propTypes = {\r\n    setTags: PropTypes.func.isRequired,\r\n};\r\n","export const getGifs = async (tags) => {\r\n    const url = `https://api.giphy.com/v1/gifs/search?q=${encodeURI(\r\n        tags\r\n    )}&limit=6&api_key=wYSQ9FXFZIjyOfD57moAlX5dTOuMLzsQ`;\r\n    const resp = await fetch(url);\r\n    const { data } = await resp.json();\r\n\r\n    const gif = data.map((img) => {\r\n        return {\r\n            id: img.id,\r\n            title: img.title,\r\n            url: img.images.downsized_medium.url,\r\n        };\r\n    });\r\n\r\n    return gif;\r\n};\r\n","import React from \"react\";\r\n\r\nexport const GifGridItem = ({ url, title, id }) => {\r\n    //En los parametros lo vuelvo a desestructurar.\r\n\r\n    return (\r\n        <>\r\n            <img\r\n                className=\"GifGrid animate__animated animate__bounce animate__bounceInRight\"\r\n                src={url}\r\n                alt={id}\r\n            />\r\n        </>\r\n    );\r\n};\r\n","import React, { useEffect, useState } from \"react\";\r\nimport { useFetchGifs } from \"../hooks/useFetchGifs\";\r\nimport { GifGridItem } from \"./GifGridItem\";\r\n\r\nexport const GifGrid = ({ tags }) => {\r\n    const { data } = useFetchGifs(tags);\r\n\r\n    return (\r\n        <>\r\n            <h3>{tags}</h3>\r\n\r\n            {data.map(({ id, title, url }) => {\r\n                return <GifGridItem key={id} url={url} title={title} />;\r\n            })}\r\n        </>\r\n    );\r\n};\r\n","import { useEffect, useState } from \"react\";\r\nimport { getGifs } from \"../helpers/getGifs\";\r\n\r\nexport const useFetchGifs = (tags) => {\r\n    const [state, setState] = useState({\r\n        data: [],\r\n        loading: true,\r\n    });\r\n\r\n    useEffect(() => {\r\n        getGifs(tags).then((imgs) => {\r\n            setState({\r\n                data: imgs,\r\n                loading: false,\r\n            });\r\n        });\r\n    }, [tags]);\r\n\r\n    return state;\r\n};\r\n","import React, { useState } from \"react\";\r\nimport { AddCategory } from \"./components/AddCategory\";\r\nimport { GifGrid } from \"./components/GifGrid\";\r\n/*\r\nconst apiKey = \"wYSQ9FXFZIjyOfD57moAlX5dTOuMLzsQ\";\r\nconst peticion = fetch(\r\n    `https://api.giphy.com/v1/gifs/random?api_key=${apiKey}`\r\n);\r\n*/\r\n\r\nexport default function App() {\r\n    const [tags, setTags] = useState([]);\r\n\r\n    return (\r\n        <>\r\n            <h2>GifExpertApp</h2>\r\n            <AddCategory setTags={setTags} />\r\n            <hr />\r\n\r\n            <ol>\r\n                {tags.map((t) => (\r\n                    <GifGrid key={t} tags={t} />\r\n                ))}\r\n            </ol>\r\n        </>\r\n    );\r\n}\r\n","import React from \"react\";\nimport ReactDOM from \"react-dom\";\nimport App from \"./gifExpertApp\";\nimport \"./index.css\";\n\nReactDOM.render(\n    <React.StrictMode>\n        <App />\n    </React.StrictMode>,\n    document.getElementById(\"root\")\n);\n"],"sourceRoot":""}