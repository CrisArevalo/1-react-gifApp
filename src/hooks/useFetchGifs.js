import { useEffect, useState } from "react";
import { getGifs } from "../helpers/getGifs";

export const useFetchGifs = (tags) => {
    const [state, setState] = useState({
        data: [],
        loading: true,
    });

    useEffect(() => {
        getGifs(tags).then((imgs) => {
            setState({
                data: imgs,
                loading: false,
            });
        });
    }, [tags]);

    return state;
};
