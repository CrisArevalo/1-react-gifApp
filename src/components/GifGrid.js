import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";
import PropTypes from "prop-types";

export const GifGrid = ({ tags }) => {
    //"tags" solo es la categoria del inputs
    const { data } = useFetchGifs(tags);

    return (
        <>
            <h3>{tags}</h3>
            {data.map(({ id, title, url }) => {
                return <GifGridItem key={id} url={url} title={title} />;
            })}
        </>
    );
};

GifGrid.propTypes = {
    tags: PropTypes.string.isRequired,
};
