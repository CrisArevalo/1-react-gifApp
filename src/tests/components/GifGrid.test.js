import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid.js";
import { useFetchGifs } from "../../hooks/useFetchGifs.js";
jest.mock("../../hooks/useFetchGifs.js");

describe("Tests de <GifGrid/>", () => {
    const tag = "Konosuba";

    // ============ TESTS ============ //
    /*
    test("Snapshot + Mock", () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });

        const wrapper = shallow(<GifGrid tags={tag} />);
        expect(wrapper).toMatchSnapshot();
    });

    test("Test useFetchGifs", () => {
        const gif = [
            {
                id: "123",
                url: "https://wallpapercave.com/wp/wp5942828.jpg",
                title: "Megumin",
            },
        ];

        useFetchGifs.mockReturnValue({
            data: gif,
            loading: false,
        });
        //El mockReturnValue simula la entrega de datos
        //como si el useFetch los hubiese entregado.
        //Para esto es necesario la linea 6.

        const wrapper = shallow(<GifGrid tags={tag} />);
        expect(wrapper).toMatchSnapshot();
    });
    */
});
