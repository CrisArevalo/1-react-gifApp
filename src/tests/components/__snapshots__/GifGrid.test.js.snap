// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests de <GifGrid/> Snapshot + Mock 1`] = `
<Fragment>
  <h3>
    Konosuba
  </h3>
</Fragment>
`;

exports[`Tests de <GifGrid/> Test useFetchGifs 1`] = `
<Fragment>
  <h3>
    Konosuba
  </h3>
  <GifGridItem
    key="123"
    title="Megumin"
    url="https://wallpapercave.com/wp/wp5942828.jpg"
  />
</Fragment>
`;
